src/librustc_trans/trans/mir/block.rs

```diff
@@ -314,6 +338,31 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
         bcx
     }
 
+    /// Create prologue cleanuppad instruction under MSVC SEH handling scheme.
+    ///
+    /// Also handles setting some state for the original trans and creating an operand bundle for
+    /// function calls.
+    fn make_cleanup_pad(&mut self, bb: mir::BasicBlock) -> Option<(ValueRef, OperandBundleDef)> {
+        let bcx = self.bcx(bb);
+        let data = self.mir.basic_block_data(bb);
+        let use_funclets = base::wants_msvc_seh(bcx.sess()) && data.is_cleanup;
+        let cleanup_pad = if use_funclets {
+            bcx.set_personality_fn(self.fcx.eh_personality());
+            Some(bcx.cleanup_pad(None, &[]))
+        } else {
+            None
+        };
+        // Set the landingpad global-state for old translator, so it knows about the SEH used.
+        bcx.set_lpad(if use_funclets {
+            Some(common::LandingPad::msvc(cleanup_pad.unwrap()))
```

Comment:
I'd personally try to structure code to avoid `.unwrap()` wherever possible, but that may just be a personal style of mine